### ISSUE TITLE
Master instruction tracker

### ISSUE THREAD

**Issue by goblinoats** [tonk-gg/dappicom#6]
This is a master tracker for completing instructions. 

## How to contribute an instruction

If you see an instruction unchecked and want to take it on please do the following:

1. Open an issue where the title of the issue is the instruction
2. Provide a description of the instruction functionality
- Reference the [nesdev instructions](https://www.nesdev.org/obelisk-6502-guide/instructions.html) wiki if possible
- Some instruction are unofficial and you may find them [here](https://www.nesdev.org/wiki/CPU_unofficial_opcodes) but the best place to look is in the [emulator/src/cpu/instr.rs](https://github.com/tonk-gg/dappicom/blob/main/emulator/src/cpu/instr.rs) file.
3. Make it clear that you will be taking this instruction on
4. Once implemented, create a pull request according to the contribution guidelines
5. Reference this issue and the issue you opened in the pull request
6. Add tests to the instruction. (All instruction files are stubbed with their inputs and expected outputs and format of the trace is explained in the notes folder.)

You may reference [How to Write an Instruction](https://github.com/tonk-gg/dappicom/blob/main/notes/how_to_write_an_instruction.md) for a brief walkthrough on how to think about writing opcodes. If you have questions, you may ask them in the discord or in the issue itself. Common questions will be collated into documentation.

Once the implementation is reviewed and the test looks good it will be merged in and the instruction will be checked off.

---
**Good for starters**

---
- [ ] clc
- [ ] cld
- [ ] cli
- [ ] clv
- [ ] dec
- [ ] dex
- [ ] dey
- [ ] eor
- [ ] inc
- [x] inx
- [ ] iny
- [ ] lda
- [ ] ldx
- [ ] ldy
- [ ] ora
- [ ] sta
- [ ] stx
- [ ] sty
- [ ] sxa
- [ ] sya
- [ ] tax
- [ ] tay
- [ ] txa
- [ ] tya

---
**For intermediate**
(These instructions have more diverse address modes)

---
- [ ] adc
- [ ] cmp
- [ ] cpx
- [ ] cpy
- [ ] lsr
- [ ] asl
- [ ] rol
- [ ] ror
- [ ] bit
- [ ] sbc

---
**More involved**
(illegal instr, stack operations, jmps, branches, system functions)

Some of the illegal instructions are simple, but we are leaving them here for now to fill those out last.

---

- [ ] alr
- [ ] anc
- [ ] and
- [ ] arr
- [ ] ahx
- [ ] axs
- [ ] dcp
- [ ] pha
- [ ] php
- [ ] pla
- [ ] ign
- [ ] isb
- [ ] plp
- [ ] tsx
- [ ] txs
- [ ] tas
- [ ] nop
- [ ] rla
- [ ] rra
- [ ] rti
- [ ] rts
- [ ] sax
- [ ] sec
- [ ] sed
- [ ] sei
- [ ] skb
- [ ] slo
- [ ] sre
- [ ] jsr
- [ ] jmp
- [ ] bcc
- [ ] bcs
- [ ] beq
- [ ] bmi 
- [ ] bne 
- [ ] bpl 
- [ ] branch 
- [ ] brk 
- [ ] bvc
- [ ] bvs
- [ ] las
- [ ] lax
- [ ] xaa
- [ ] xxx

**Comment by ameya-deshmukh** [comment]
@goblinoats looking for contributors to these? Would love to take a stab at some

**Comment by goblinoats** [comment]
@ameya-deshmukh Yeah! Take your pic of which instruction you'd like to take on and then just open a separate issue for it so that we know it's in progress and that you've claimed it. Lmk if you run into any speed bumps. Some of the instructions will require more thought than others.

**Comment by ameya-deshmukh** [comment]
Got it! Thanks, will let you know my pick

**Comment by TAdev0** [comment]
@baz-lightyear  @goblinoats  hey :) I would like to implement one of these opcodes to get started, is one of them especially easy for a first contribution? Thank you!

**Comment by goblinoats** [comment]
@TAdev0 You could take a look at iny or dex/dey as they are very similar to inx (which already has an example implementation). The simple operations like binary operations and arithmetic operations are the easiest ones to reason about.

**Comment by TAdev0** [comment]
Thank you @goblinoats. I ll work on these !

**Comment by yehia67** [comment]
Hi @goblinoats, I would like to help😄 Which opcode should I start with? 👀

**Comment by goblinoats** [comment]
Hey @yehia67, I've sorted the instructions now into categories.

**Comment by TAdev0** [comment]
@goblinoats  should i rename the `check_op` function as `main` function just like INY after implementing the opcode?

And where do these values come from in the INX example?

```

//PC = 8203
//X = 8201
//Y = 8202
//A = 8200
//SR = 8205
```

thank you!

**Comment by cryptogru725** [comment]
May I take this issue on?

**Comment by chiscookeke11** [comment]
Could I be assigned to this?

**Comment by krishnabandewar** [comment]
Could I take on this issue?
